src/variant.rs

```diff
@@ -141,6 +142,19 @@ impl Variant {
             }
         }
     }
+
+    /// Constructs a new serialised-mode GVariant instance.
+    pub fn new_from_bytes<T: StaticVariantType>(bytes: &Bytes, trusted: bool) -> Self {
```

Comment:
If you pass random data and `trusted=true`, will it crash? If so this needs to be an `unsafe fn` (you probably want a safe variant with `trusted=false` and an unsafe with `trusted=true` then).

Reply:
I'll check.

Reply:
i tried a few things (mainly tried to give it less bytes and some curropted bytes) with `trusted=true` but i failed to make it crash. Also looking briefly through the code and docs, it seems unlikely that it skips critical checks when decoding the bytes so I think it's good like this.

Reply:
What is it actually doing then? I feel like this should get an issue reported for improving the docs against GLib :)

Reply:
Actually ebassi says it's UB so I'll go with your approach.